metrics/firebase/test/firestore/rules/feature-config-test.js

```diff
@@ -0,0 +1,211 @@
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Authenticated with a password and allowed email domain user with not verified email',
+      'app': await getApplicationWith(
+        getAllowedEmailUser(passwordSignInProviderId, false)
+      ),
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Authenticated with a password and not allowed email domain user with not verified email',
+      'app': await getApplicationWith(
+        getDeniedEmailUser(passwordSignInProviderId, false)
+      ),
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Authenticated with google and allowed email domain user with a verified email',
+      'app': await getApplicationWith(
+        getAllowedEmailUser(googleSignInProviderId, true)
+      ),
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Authenticated with google and not allowed email domain user with a verified email',
+      'app': await getApplicationWith(
+        getDeniedEmailUser(googleSignInProviderId, true)
+      ),
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Authenticated with google and allowed email domain user with not verified email',
+      'app': await getApplicationWith(
+        getAllowedEmailUser(googleSignInProviderId, false)
+      ),
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Authenticated with google and not allowed email domain user with not verified email',
+      'app': await getApplicationWith(
+        getDeniedEmailUser(googleSignInProviderId, false)
+      ),
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+    {
+      'describe': 'Unauthenticated user',
+      'app': unauthenticatedApp,
+      'can': {
+        'create': false,
+        'read': true,
+        'update': false,
+        'delete': false,
+      }
+    },
+  ];
+
+  before(async () => {
+    await setupTestDatabaseWith(
+      Object.assign({}, featureConfig, allowedEmailDomains)
+    );
+  });
+
+  describe("Feature config collection rules", () => {
+    async.forEach(users, (user, callback) => {
+      describe(user.describe, function () {
+        let canCreateDescription = user.can.create
+          ? "allows to create an feature config"
+          : "does not allow creating an feature config";
+        let canReadDescription = user.can.read
+          ? "allows reading an feature config"
+          : "does not allow reading an feature configs";
```

Comment:
```suggestion
          ? "allows reading a feature config"
          : "does not allow reading a feature configs";
```
Please apply to all test cases.